### PR TITLE
subversion: fix darwin

### PR DESCRIPTION
**Pull request by khaneliman**
## Description of changes
Getting errors building git with svn because of subversion-client failing on darwin. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by khaneliman** [comment]
Result of `nixpkgs-review pr 341232` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by khaneliman** [comment]
Result of `nixpkgs-review pr 341232` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.pysvn</li>
    <li>python312Packages.pysvn</li>
  </ul>
</details>
<details>
  <summary>36 packages built:</summary>
  <ul>
    <li>apacheHttpdPackages.subversion</li>
    <li>apacheHttpdPackages.subversion.dev</li>
    <li>apacheHttpdPackages.subversion.man</li>
    <li>cabal2nix</li>
    <li>git-doc (git-doc.doc ,gitFull.doc)</li>
    <li>gitFull</li>
    <li>gitSVN</li>
    <li>gitSVN.doc</li>
    <li>luarocks-packages-updater</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-prefetch-svn</li>
    <li>nix-update-source</li>
    <li>nix-update-source.dist</li>
    <li>patroni</li>
    <li>patroni.dist</li>
    <li>perl536Packages.SVNSimple</li>
    <li>perl536Packages.SVNSimple.devdoc</li>
    <li>perl538Packages.SVNSimple</li>
    <li>perl538Packages.SVNSimple.devdoc</li>
    <li>python311Packages.ydiff</li>
    <li>python311Packages.ydiff.dist</li>
    <li>ydiff (python312Packages.ydiff)</li>
    <li>ydiff.dist (python312Packages.ydiff.dist)</li>
    <li>rapidsvn</li>
    <li>subversion</li>
    <li>subversion.dev</li>
    <li>subversion.man</li>
    <li>subversionClient</li>
    <li>subversionClient.dev</li>
    <li>subversionClient.man</li>
    <li>svn-all-fast-export</li>
    <li>svn2git</li>
    <li>svnfs</li>
    <li>vcstool</li>
    <li>vcstool.dist</li>
    <li>wp4nix</li>
  </ul>
</details>

File: pkgs/applications/version-management/subversion/default.nix [hunk 1]
```diff
@@ -49,6 +49,14 @@ let
     # "-P" CPPFLAG is needed to build Python bindings and subversionClient
     CPPFLAGS = [ "-P" ];
 
+    env = lib.optionalAttrs stdenv.cc.isClang {
+      NIX_CFLAGS_COMPILE = toString [
```
Comment (emilazy) [inline pkgs/applications/version-management/subversion/default.nix:53]: I forget why but IIRC `toString` on lists is cursed; best to explicitly `concatStringsSep " "`.

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 341232` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.pysvn</li>
    <li>python312Packages.pysvn</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>patroni</li>
    <li>patroni.dist</li>
  </ul>
</details>
<details>
  <summary>34 packages built:</summary>
  <ul>
    <li>apacheHttpdPackages.subversion</li>
    <li>apacheHttpdPackages.subversion.dev</li>
    <li>apacheHttpdPackages.subversion.man</li>
    <li>cabal2nix</li>
    <li>git-doc (git-doc.doc ,gitFull.doc)</li>
    <li>gitFull</li>
    <li>gitSVN</li>
    <li>gitSVN.doc</li>
    <li>luarocks-packages-updater</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-prefetch-svn</li>
    <li>nix-update-source</li>
    <li>nix-update-source.dist</li>
    <li>perl536Packages.SVNSimple</li>
    <li>perl536Packages.SVNSimple.devdoc</li>
    <li>perl538Packages.SVNSimple</li>
    <li>perl538Packages.SVNSimple.devdoc</li>
    <li>python311Packages.ydiff</li>
    <li>python311Packages.ydiff.dist</li>
    <li>ydiff (python312Packages.ydiff)</li>
    <li>ydiff.dist (python312Packages.ydiff.dist)</li>
    <li>rapidsvn</li>
    <li>subversion</li>
    <li>subversion.dev</li>
    <li>subversion.man</li>
    <li>subversionClient</li>
    <li>subversionClient.dev</li>
    <li>subversionClient.man</li>
    <li>svn-all-fast-export</li>
    <li>svn2git</li>
    <li>svnfs</li>
    <li>vcstool</li>
    <li>vcstool.dist</li>
    <li>wp4nix</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
Result of `nixpkgs-review pr 341232` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.pysvn</li>
    <li>python312Packages.pysvn</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>patroni</li>
    <li>patroni.dist</li>
  </ul>
</details>
<details>
  <summary>34 packages built:</summary>
  <ul>
    <li>apacheHttpdPackages.subversion</li>
    <li>apacheHttpdPackages.subversion.dev</li>
    <li>apacheHttpdPackages.subversion.man</li>
    <li>cabal2nix</li>
    <li>git-doc (git-doc.doc ,gitFull.doc)</li>
    <li>gitFull</li>
    <li>gitSVN</li>
    <li>gitSVN.doc</li>
    <li>luarocks-packages-updater</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-prefetch-svn</li>
    <li>nix-update-source</li>
    <li>nix-update-source.dist</li>
    <li>perl536Packages.SVNSimple</li>
    <li>perl536Packages.SVNSimple.devdoc</li>
    <li>perl538Packages.SVNSimple</li>
    <li>perl538Packages.SVNSimple.devdoc</li>
    <li>python311Packages.ydiff</li>
    <li>python311Packages.ydiff.dist</li>
    <li>ydiff (python312Packages.ydiff)</li>
    <li>ydiff.dist (python312Packages.ydiff.dist)</li>
    <li>rapidsvn</li>
    <li>subversion</li>
    <li>subversion.dev</li>
    <li>subversion.man</li>
    <li>subversionClient</li>
    <li>subversionClient.dev</li>
    <li>subversionClient.man</li>
    <li>svn-all-fast-export</li>
    <li>svn2git</li>
    <li>svnfs</li>
    <li>vcstool</li>
    <li>vcstool.dist</li>
    <li>wp4nix</li>
  </ul>
</details>

**Review by Aleksanaa — APPROVED** [review]

**Comment by viraptor** [comment]
Result of `nixpkgs-review pr 341232` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.pysvn</li>
    <li>python312Packages.pysvn</li>
  </ul>
</details>
<details>
  <summary>36 packages built:</summary>
  <ul>
    <li>apacheHttpdPackages.subversion</li>
    <li>apacheHttpdPackages.subversion.dev</li>
    <li>apacheHttpdPackages.subversion.man</li>
    <li>cabal2nix</li>
    <li>git-doc (git-doc.doc ,gitFull.doc)</li>
    <li>gitFull</li>
    <li>gitSVN</li>
    <li>gitSVN.doc</li>
    <li>luarocks-packages-updater</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-prefetch-svn</li>
    <li>nix-update-source</li>
    <li>nix-update-source.dist</li>
    <li>patroni</li>
    <li>patroni.dist</li>
    <li>perl536Packages.SVNSimple</li>
    <li>perl536Packages.SVNSimple.devdoc</li>
    <li>perl538Packages.SVNSimple</li>
    <li>perl538Packages.SVNSimple.devdoc</li>
    <li>python311Packages.ydiff</li>
    <li>python311Packages.ydiff.dist</li>
    <li>ydiff (python312Packages.ydiff)</li>
    <li>ydiff.dist (python312Packages.ydiff.dist)</li>
    <li>rapidsvn</li>
    <li>subversion</li>
    <li>subversion.dev</li>
    <li>subversion.man</li>
    <li>subversionClient</li>
    <li>subversionClient.dev</li>
    <li>subversionClient.man</li>
    <li>svn-all-fast-export</li>
    <li>svn2git</li>
    <li>svnfs</li>
    <li>vcstool</li>
    <li>vcstool.dist</li>
    <li>wp4nix</li>
  </ul>
</details>

**Comment by techknowlogick** [comment]
Result of `nixpkgs-review pr 341232` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.pysvn</li>
    <li>python312Packages.pysvn</li>
  </ul>
</details>
<details>
  <summary>36 packages built:</summary>
  <ul>
    <li>apacheHttpdPackages.subversion</li>
    <li>apacheHttpdPackages.subversion.dev</li>
    <li>apacheHttpdPackages.subversion.man</li>
    <li>cabal2nix</li>
    <li>git-doc (git-doc.doc ,gitFull.doc)</li>
    <li>gitFull</li>
    <li>gitSVN</li>
    <li>gitSVN.doc</li>
    <li>luarocks-packages-updater</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-prefetch-svn</li>
    <li>nix-update-source</li>
    <li>nix-update-source.dist</li>
    <li>patroni</li>
    <li>patroni.dist</li>
    <li>perl536Packages.SVNSimple</li>
    <li>perl536Packages.SVNSimple.devdoc</li>
    <li>perl538Packages.SVNSimple</li>
    <li>perl538Packages.SVNSimple.devdoc</li>
    <li>python311Packages.ydiff</li>
    <li>python311Packages.ydiff.dist</li>
    <li>ydiff (python312Packages.ydiff)</li>
    <li>ydiff.dist (python312Packages.ydiff.dist)</li>
    <li>rapidsvn</li>
    <li>subversion</li>
    <li>subversion.dev</li>
    <li>subversion.man</li>
    <li>subversionClient</li>
    <li>subversionClient.dev</li>
    <li>subversionClient.man</li>
    <li>svn-all-fast-export</li>
    <li>svn2git</li>
    <li>svnfs</li>
    <li>vcstool</li>
    <li>vcstool.dist</li>
    <li>wp4nix</li>
  </ul>
</details>

**Review by viraptor — APPROVED** [review]

**Comment by khaneliman** [comment]
any chance we could get a merge on this?

**Pull request merged**